pkg/server/rpc_server.go

```diff
@@ -239,6 +240,9 @@ func (s *rpcServer) createSession() (session.Session, error) {
 		action := &memory.PanicOnExceed{Killer: &vars.SQLKiller}
 		vars.MemTracker.SetActionOnExceed(action)
 	}
+	if err = vars.SetSystemVar(variable.MaxAllowedPacket, strconv.FormatUint(variable.DefMaxAllowedPacket, 10)); err != nil {
```

Comment:
Added this line to make sure the value `variable.DefMaxAllowedPacket` is present to make test pass. The old code is using `mock.Context` with `DefMaxAllowedPacket` set inner `ScalarFunction`. But we use the same sessioncontext object in this PR.